### PR TITLE
4394 by Inlead: Place2book: Clone of events does not create new ticket.

### PR DESCRIPTION
**Pull request by NicolaeGrigoriu**
#### Link to issue

https://platform.dandigbib.org/issues/4394

#### Description

_Solution_

On the clone stage(node/{node id}/clone/confirm page) form submit it is copied the old event from the p2b and are removed all relations(id of event and prices ids).

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

### CODE REVIEW

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 331b87c0ea057e747b712a05450cc541962e9982
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 331b87c0ea057e747b712a05450cc541962e9982
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by holt83** [comment]
Isnt't this a bit hacky? On the node clone module description page it mentions a hook which can be used if we want to modify the cloned node: https://www.drupal.org/project/node_clone

Have you tried using this?

**Comment by NicolaeGrigoriu** [comment]
I agree that this solution is not elegant. When the _prepopulate_ method is used then the `hook_clone_node_alter` is triggered before `hook_field_widget_form`. In this case, when unset the relation on the clone_node_alter the prices form will be empty. If the _save-edit_ method is used then the prices will be attached to the original event and will required another hack to clear the things up.

File: modules/ding_place2book/ding_place2book.fields.inc [hunk 1]
```diff
@@ -566,9 +566,19 @@ function _ding_place2book_sync_p2b_entities($entity, array $settings) {
   $items = [];
 
   $event_id = !empty($settings['event_id']) ? $settings['event_id'] : FALSE;
+  $prices = $settings['prices_wrapper']['prices'];
   $event_maker_id = FALSE;
+
   // For clarity.
   $is_new = !$event_id;
+  if (arg(2) == 'clone') {
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.fields.inc]: Having a dependency on the page url deep down in a internal function does not seem like a good idea.

Is there something we can do using the clone api e.g. [`hook_clone_node_alter`](https://git.drupalcode.org/project/node_clone/blob/7.x-1.x/clone.api.php#L22)?

Comment (NicolaeGrigoriu) [reply modules/ding_place2book/ding_place2book.fields.inc]: The `hook_clone_node_alter` is called before the node form is rendered. As it is working now it is not possible to use this hook. As a solution we can store event id in a hidden form field and use it when make field presave(the POST/PUT/DELETE requests). This field will be populated on the node edit form. And on node clone this id will be removed so, the prices will be attached to the newly created entity.

**Comment by holt83** [comment]
I have made an alternative solution using hook_node_clone_alter() that also takes care of removing event ID in: https://platform.dandigbib.org/issues/4394

**Comment by kasperg** [comment]
Thanks both of you. I prefer the solution in #1488 and am closing this in favour of it.

**Pull request closed without merging**